measurements/api/measurements.py

```diff
@@ -206,41 +221,43 @@ def get_measurement(measurement_id, download=None):
         raise BadRequest("Invalid measurement_id")
     msm_no = int(m.group(1))
 
-    q = (
-        current_app.db_session.query(
-            Measurement.report_no.label("report_no"),
-            Measurement.frame_off.label("frame_off"),
-            Measurement.frame_size.label("frame_size"),
-            Measurement.intra_off.label("intra_off"),
-            Measurement.intra_size.label("intra_size"),
-            Report.textname.label("textname"),
-            Report.report_no.label("r_report_no"),
-            Report.autoclaved_no.label("r_autoclaved_no"),
-            Autoclaved.filename.label("a_filename"),
-            Autoclaved.autoclaved_no.label("a_autoclaved_no"),
+    cols = [
+        literal_column("measurement.report_no"),
+        literal_column("frame_off"),
+        literal_column("frame_size"),
+        literal_column("intra_off"),
+        literal_column("intra_size"),
+        literal_column("textname"),
+        literal_column("report.autoclaved_no"),
+        literal_column("autoclaved.filename"),
+    ]
+    table = (
+        sql.table("measurement")
+        .join(
+            sql.table("report"), sql.text("measurement.report_no = report.report_no"),
         )
-        .filter(Measurement.msm_no == msm_no)
-        .join(Report, Report.report_no == Measurement.report_no)
-        .join(Autoclaved, Autoclaved.autoclaved_no == Report.autoclaved_no)
-    )
-    try:
-        msmt = q.one()
-    except MultipleResultsFound:
-        current_app.logger.warning(
-            "Duplicate rows for measurement_id: %s" % measurement_id
+        .join(
+            sql.table("autoclaved"),
+            sql.text("autoclaved.autoclaved_no = report.autoclaved_no"),
         )
-        msmt = q.first()
-    except NoResultFound:
-        # XXX we should actually return a 404 here
-        raise BadRequest("No measurement found")
+    )
+    where = sql.text("measurement.msm_no = :msm_no")
+    query = select(cols).where(where).select_from(table)
+    query_params = dict(msm_no=msm_no)
+    q = current_app.db_session.execute(query, query_params)
+
+    msmt = q.fetchone()
```

Comment:
Should be noted we aren't handling the case in which there are duplicate results. We may or may not care about that.

In theory it's easy to fix by just doing an additional call to `fetchone()` and ensuring we get `None` on the second call.

Reply:
Until we introduce OOID we cannot pinpoint msmts and get_measurement is simply returning the row.